File: futbolfan1.client/src/app/model/player.ts
// src/app/models/player.model.ts

export interface Player {
  id: number;
  name: string;
  position: string;
  age: number;
  cost: number;
  salary: number;
  role?: string;  // Ruolo del giocatore, opzionale
  isStarting: boolean;  // Se il giocatore è titolare
  teamId?: number;  // ID della squadra a cui appartiene (opzionale)

  // Statistiche del giocatore
  speed: number;
  shooting: number;
  passing: number;
  dribbling: number;
  defense: number;
  physical: number;

  // Overall rating
  overall: number;  // Media delle statistiche principali
}

File: Futbolfan1.Server/Models/Player.cs
using System.Text.Json.Serialization;

namespace FutbolFan1.Models
{
    public class Player
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Position { get; set; }
        public int Age { get; set; }
        public decimal Cost { get; set; }
        public decimal Salary { get; set; }
        public string Role { get; set; }
        public bool IsStarting { get; set; }

        // TeamId è facoltativo, per gestire giocatori senza squadra
        public int? TeamId { get; set; }

        // Associare un giocatore a una squadra
        [JsonIgnore]  // Evita cicli di serializzazione
        public Team Team { get; set; }

        // Attributi delle abilità del giocatore
        public decimal Speed { get; set; }
        public decimal Shooting { get; set; }
        public decimal Passing { get; set; }
        public decimal Dribbling { get; set; }
        public decimal Defense { get; set; }
        public decimal Physical { get; set; }
    }
}
